processor/stream/package_tests/error_attrs_test.go

```diff
@@ -0,0 +1,212 @@
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package package_tests
+
+import (
+	"testing"
+
+	"github.com/elastic/apm-server/model/error/generated/schema"
+
+	"github.com/elastic/apm-server/processor/stream"
+	"github.com/elastic/apm-server/tests"
+)
+
+func errorProcSetup() *tests.ProcessorSetup {
+	return &tests.ProcessorSetup{
+		Proc:            &V2TestProcessor{StreamProcessor: stream.StreamProcessor{}},
+		FullPayloadPath: "../testdata/intake-v2/errors.ndjson",
+		TemplatePaths: []string{
+			"../../../model/error/_meta/fields.yml",
+			"../../../_meta/fields.common.yml",
+		},
+		Schema:       schema.ModelSchema,
+		SchemaPrefix: "error",
+	}
+}
+
+func errorPayloadAttrsNotInFields() *tests.Set {
+	return tests.NewSet(
+		tests.Group("error.exception.attributes"),
+		"error.exception.stacktrace",
+		"error.log.stacktrace",
+	)
+}
+
+func errorFieldsNotInPayloadAttrs() *tests.Set {
+	return tests.NewSet(
+		"listening", "view errors", "error id icon",
+		"context.user.user-agent", "context.user.ip", "context.system.ip",
+	)
+}
+
+func errorPayloadAttrsNotInJsonSchema() *tests.Set {
+	return tests.NewSet(
+		"error",
+		"error.log.stacktrace.vars.key",
+		"error.exception.stacktrace.vars.key",
+		"error.exception.attributes.foo",
+		"error.context.request.headers.some-other-header",
+		"error.context.request.headers.array",
+		tests.Group("error.context.custom"),
+		tests.Group("error.context.request.env"),
+		tests.Group("error.context.request.cookies"),
+		tests.Group("error.context.tags"),
+	)
+}
+
+func errorRequiredKeys() *tests.Set {
+	return tests.NewSet(
+		"error",
+		"error.id",
+		"error.exception.message",
+		"error.exception",
+		"error.log",
+		"error.log.type",
+		"error.log.message",
+		"error.exception.stacktrace.filename",
+		"error.exception.stacktrace.lineno",
+		"error.log.stacktrace.filename",
+		"error.log.stacktrace.lineno",
+		"error.context.request.method",
+		"error.context.request.url",
+
+		"error.trace_id",
+		"error.transaction_id",
+		"error.parent_id",
+	)
+}
+
+type val = []interface{}
+type obj = map[string]interface{}
+
+func errorCondRequiredKeys() map[string]tests.Condition {
+	return map[string]tests.Condition{
+		"error.exception": tests.Condition{Absence: []string{"error.log"}},
+		"error.log":       tests.Condition{Absence: []string{"error.exception"}},
+
+		"error.message": tests.Condition{Absence: []string{"error.type"}},
+		"error.type":    tests.Condition{Absence: []string{"error.message"}},
```

Comment:
This should be `error.exception.message` and `error.exception.type`. I assume the test doesn't fail as the specified attribute doesn't exist, so it never actually tests it.